python/examples/minimal_build/build_conda.sh

```diff
@@ -79,6 +79,7 @@ cmake -GNinja \
       -DCMAKE_BUILD_TYPE=DEBUG \
       -DCMAKE_INSTALL_PREFIX=$ARROW_HOME \
       -DCMAKE_INSTALL_LIBDIR=lib \
+      -DARROW_DATASET=ON \
```

Comment:
I added it as a temporal workaround for the issue solved on this PR: [ARROW-16526](https://github.com/apache/arrow/pull/13116)
As this has been fixed I will revert it. I was planning to also remove `PARQUET` from the minimal build as we should have a build without optional dependencies. What do you think about removing that?

Reply:
Yes, I think we should actually remove Parquet and (especially) Flight.

Reply:
I have removed Dataset, Parquet and Flight. The builds are now failing because there seems to be a test which is not being marked to be skipped when parquet is not available:
```
     def test_parquet_invalid_version(tempdir):
        table = pa.table({'a': [1, 2, 3]})
        with pytest.raises(ValueError, match="Unsupported Parquet format version"):
>           _write_table(table, tempdir / 'test_version.parquet', version="2.2")
E           NameError: name '_write_table' is not defined
```

Reply:
That test probably deserves fixing then ;-)

Reply:
Now that the PR to fix the parquet pytest marks have been merged I have rebased and we can see how the minimal builds without Dataset, Parquet and Flight are successful again.